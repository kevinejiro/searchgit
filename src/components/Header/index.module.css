.header {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: none;
  z-index: 4;
  background: transparent no-repeat padding-box;
  position: absolute;
  top: 0;
  margin: 3.1rem 0rem 3.6rem;
  box-sizing: border-box;
}

.headerinner {
  width: 90%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 65px;
  max-width: 1280px;
}

.theme-mode {
  top: -2rem;
  left: 2%;
  position: absolute;
  display: flex;
  align-items: center;
  column-gap: 1.6rem;
  user-select: none;
}

.moon-wrapper,
.sun-wrapper {
  cursor: pointer;
}

.search-icon-wrapper {
  display: none;
  justify-content: center;
  align-items: center;
  margin-bottom: 1%;
}

.search-icon {
  width: 25%;
}

.search-icon-text {
  font: normal normal 300 22px/30px Metropolis;
  letter-spacing: 0px;
  color: var(--input-text-color);
  margin-left: 0.4rem;
  text-decoration: none;
}

.search-icon-text > span {
  font: normal normal bold 26px/31px Metropolis;
}

/* Tablet */
@media (min-width: 768px) {
  .theme-mode {
    top: 0;
    left: 0;
    position: relative;
    display: flex;
    align-items: center;
    column-gap: 1.6rem;
    user-select: none;
  }
  .search-icon-wrapper {
    display: flex;
  }

  .header {
    /* margin: 14rem 7.85rem 3.6rem; */
  }
}

/* Desktop */
@media (min-width: 1024px) {
  .theme-mode {
    position: relative;
    display: flex;
    align-items: center;
    column-gap: 1.6rem;
    user-select: none;
  }
  .search-icon-wrapper {
    display: flex;
  }
  .header {
    /* margin: 14.4rem 0 3.6rem; */
  }
}
