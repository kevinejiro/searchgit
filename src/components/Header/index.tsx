import { useEffect } from "react";
import { Link, useLocation } from "react-router-dom";
import { motion } from "framer-motion";
import useLocalStorage from "use-local-storage";

// icons
import MoonIcon from "../common/MoonIcon";
import SunIcon from "../common/SunIcon";
import GitIcon from "../common/GitIcon";

// css import
import classes from "./index.module.css";

const transition = { duration: 0.6, ease: ["easeOut"], delay: 1.2 };

const Header: React.FC = () => {
  // Detect if the user has dark color theme.
  const prefferedDarkMode = window.matchMedia(
    "(prefers-color-scheme: dark)"
  ).matches;
  const [theme, setTheme] = useLocalStorage<string>(
    "theme",
    prefferedDarkMode ? "dark" : "light"
  );

  let location = useLocation();

  useEffect(() => {
    document.documentElement.setAttribute("theme-mode", theme);
  }, [theme]);

  const toggleThemeHandler = (mode: string) => {
    if (mode === "dark") {
      document.documentElement.setAttribute("theme-mode", "dark");
      setTheme("dark");
    } else if (mode === "light") {
      document.documentElement.setAttribute("theme-mode", "light");
      setTheme("light");
    }
  };

  return (
    <header className={classes.header} data-theme={theme}>
      <div
        className={classes.headerinner}
        style={{
          justifyContent:
            location.pathname === "/results" ? "space-between" : "flex-end",
        }}
      >
        {location.pathname === "/results" && (
          <Link
            to="/"
            style={{
              textDecoration: "none",
            }}
          >
            <motion.div
              initial={{ x: "10rem", opacity: 0 }}
              animate={{ x: "0rem", opacity: 1 }}
              transition={transition}
              className={classes["search-icon-wrapper"]}
            >
              <div className={classes["search-icon"]}>
                <GitIcon />
              </div>
              <div className={classes["search-icon-text"]}>
                search<span>Git</span>
              </div>{" "}
            </motion.div>
          </Link>
        )}

        <div
          className={`${classes["theme-mode"]} ${
            classes[`${location.pathname === "/results" ? "result" : ""}`]
          }`}
          // style={{
          //   display: location.pathname === "/results" ? "none" : "flex",
          // }}
        >
          <div
            onClick={() => toggleThemeHandler("dark")}
            className={classes["moon-wrapper"]}
            data-testid="moon-wrapper"
          >
            <MoonIcon fill={theme === "dark" ? "#27D3A5" : "#3E424B"} />
          </div>
          <div
            onClick={() => toggleThemeHandler("light")}
            data-testid="sun-wrapper"
            className={classes["sun-wrapper"]}
          >
            <SunIcon fill={theme !== "light" ? "#3E424B" : "#27D3A5"} />
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
