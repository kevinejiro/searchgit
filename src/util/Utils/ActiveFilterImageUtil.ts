import contributorsImageActive from "../../assets/images/contributorsActive.svg";
import repoActive from "../../assets/images/repoActive.svg";
import contributorsImage from "../../assets/images/contributors.svg";
import repoImage from "../../assets/images/repo.svg";

const ActiveFilterImageUtil = {
  getImageList(input: string) {
    let contributionsImageUrl;
    let repositoriesImageUrl;
    let urlList;
    switch (input) {
      case "user":
        contributionsImageUrl = contributorsImageActive;
        repositoriesImageUrl = repoImage;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
      case "org":
        contributionsImageUrl = contributorsImage;
        repositoriesImageUrl = repoActive;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
      default:
        contributionsImageUrl = contributorsImage;
        repositoriesImageUrl = repoImage;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
    }
  },
};

export default ActiveFilterImageUtil;
