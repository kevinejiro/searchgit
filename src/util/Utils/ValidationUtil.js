const ValidationUtil = {
	nullChecker(input) {
		if (input !== undefined && input !== null) {
			return input;
		}

		return "N/A";
	},
};

export default ValidationUtil;
