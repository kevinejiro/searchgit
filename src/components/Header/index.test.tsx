import { render } from "../../util/Utils/test-utils";
// import userEvent from "@testing-library/user-event";

import Header from "./index";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/results",
  }),
}));

describe("<Header />", () => {
  it("renders the component without errors", () => {
    const component = render(<Header />);
    expect(component).toMatchSnapshot();
  });
  // it("changes theme successfully", () => {
  //   render(<Header />);
  //   const toggleThemeHandler = jest.fn();
  //   userEvent.click(screen.getByTestId("moon-wrapper"));
  //   expect(toggleThemeHandler).toBeCalledWith("dark");
  // });
});
