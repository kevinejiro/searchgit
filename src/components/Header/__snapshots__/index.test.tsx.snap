// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the component without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
        data-theme="light"
      >
        <div
          class="headerinner"
          style="justify-content: flex-end;"
        >
          <div
            class="theme-mode"
          >
            <div
              class="moon-wrapper"
              data-testid="moon-wrapper"
            >
              <div
                class="svgWrapper"
              >
                <svg
                  class="svg"
                  height="20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.513 11.397a.701.701 0 00-.588.128 7.496 7.496 0 01-2.276 1.336 7.101 7.101 0 01-2.583.462 7.505 7.505 0 01-5.32-2.209 7.568 7.568 0 01-2.199-5.342c0-.873.154-1.72.41-2.49a6.904 6.904 0 011.227-2.21.657.657 0 00-.102-.924.701.701 0 00-.589-.128C5.32.61 3.427 1.92 2.072 3.666A10.158 10.158 0 000 9.83c0 2.8 1.125 5.342 2.967 7.19a10.025 10.025 0 007.16 2.98c2.353 0 4.527-.822 6.266-2.183a10.13 10.13 0 003.58-5.624.623.623 0 00-.46-.796z"
                    fill="#3E424B"
                    fill-rule="nonzero"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sun-wrapper"
              data-testid="sun-wrapper"
            >
              <div
                class="svgWrapper"
              >
                <svg
                  class="svg"
                  height="20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#27D3A5"
                    fill-rule="nonzero"
                  >
                    <path
                      d="M13.545 6.455c-.9-.9-2.17-1.481-3.545-1.481a4.934 4.934 0 00-3.545 1.481c-.9.9-1.481 2.17-1.481 3.545 0 1.376.582 2.646 1.481 3.545.9.9 2.17 1.481 3.545 1.481a4.934 4.934 0 003.545-1.481c.9-.9 1.481-2.17 1.481-3.545a4.934 4.934 0 00-1.481-3.545zM10 3.413a.7.7 0 00.688-.688V.688A.7.7 0 0010 0a.7.7 0 00-.688.688v2.037a.7.7 0 00.688.688zM15.635 5.344l1.455-1.455a.67.67 0 000-.952.67.67 0 00-.952 0l-1.455 1.455a.67.67 0 000 .952c.238.264.66.264.952 0zM19.312 9.312h-2.037a.7.7 0 00-.688.688.7.7 0 00.688.688h2.037A.7.7 0 0020 10a.7.7 0 00-.688-.688zM15.608 14.656a.67.67 0 00-.952 0 .67.67 0 000 .952l1.455 1.455a.67.67 0 00.952 0 .67.67 0 000-.952l-1.455-1.455zM10 16.587a.7.7 0 00-.688.688v2.037A.7.7 0 0010 20a.7.7 0 00.688-.688v-2.037a.7.7 0 00-.688-.688zM4.365 14.656L2.91 16.111a.67.67 0 000 .952.67.67 0 00.952 0l1.455-1.455a.67.67 0 000-.952c-.238-.264-.66-.264-.952 0zM3.413 10a.7.7 0 00-.688-.688H.688A.7.7 0 000 10a.7.7 0 00.688.688h2.037A.7.7 0 003.413 10zM4.365 5.344a.67.67 0 00.952 0 .67.67 0 000-.952L3.862 2.937a.67.67 0 00-.952 0 .67.67 0 000 .952l1.455 1.455z"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
      data-theme="light"
    >
      <div
        class="headerinner"
        style="justify-content: flex-end;"
      >
        <div
          class="theme-mode"
        >
          <div
            class="moon-wrapper"
            data-testid="moon-wrapper"
          >
            <div
              class="svgWrapper"
            >
              <svg
                class="svg"
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.513 11.397a.701.701 0 00-.588.128 7.496 7.496 0 01-2.276 1.336 7.101 7.101 0 01-2.583.462 7.505 7.505 0 01-5.32-2.209 7.568 7.568 0 01-2.199-5.342c0-.873.154-1.72.41-2.49a6.904 6.904 0 011.227-2.21.657.657 0 00-.102-.924.701.701 0 00-.589-.128C5.32.61 3.427 1.92 2.072 3.666A10.158 10.158 0 000 9.83c0 2.8 1.125 5.342 2.967 7.19a10.025 10.025 0 007.16 2.98c2.353 0 4.527-.822 6.266-2.183a10.13 10.13 0 003.58-5.624.623.623 0 00-.46-.796z"
                  fill="#3E424B"
                  fill-rule="nonzero"
                />
              </svg>
            </div>
          </div>
          <div
            class="sun-wrapper"
            data-testid="sun-wrapper"
          >
            <div
              class="svgWrapper"
            >
              <svg
                class="svg"
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#27D3A5"
                  fill-rule="nonzero"
                >
                  <path
                    d="M13.545 6.455c-.9-.9-2.17-1.481-3.545-1.481a4.934 4.934 0 00-3.545 1.481c-.9.9-1.481 2.17-1.481 3.545 0 1.376.582 2.646 1.481 3.545.9.9 2.17 1.481 3.545 1.481a4.934 4.934 0 003.545-1.481c.9-.9 1.481-2.17 1.481-3.545a4.934 4.934 0 00-1.481-3.545zM10 3.413a.7.7 0 00.688-.688V.688A.7.7 0 0010 0a.7.7 0 00-.688.688v2.037a.7.7 0 00.688.688zM15.635 5.344l1.455-1.455a.67.67 0 000-.952.67.67 0 00-.952 0l-1.455 1.455a.67.67 0 000 .952c.238.264.66.264.952 0zM19.312 9.312h-2.037a.7.7 0 00-.688.688.7.7 0 00.688.688h2.037A.7.7 0 0020 10a.7.7 0 00-.688-.688zM15.608 14.656a.67.67 0 00-.952 0 .67.67 0 000 .952l1.455 1.455a.67.67 0 00.952 0 .67.67 0 000-.952l-1.455-1.455zM10 16.587a.7.7 0 00-.688.688v2.037A.7.7 0 0010 20a.7.7 0 00.688-.688v-2.037a.7.7 0 00-.688-.688zM4.365 14.656L2.91 16.111a.67.67 0 000 .952.67.67 0 00.952 0l1.455-1.455a.67.67 0 000-.952c-.238-.264-.66-.264-.952 0zM3.413 10a.7.7 0 00-.688-.688H.688A.7.7 0 000 10a.7.7 0 00.688.688h2.037A.7.7 0 003.413 10zM4.365 5.344a.67.67 0 00.952 0 .67.67 0 000-.952L3.862 2.937a.67.67 0 00-.952 0 .67.67 0 000 .952l1.455 1.455z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
